.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,44 @@
+name: Run tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  run-unit-tests:
+      name: Run unit tests
+      runs-on: ubuntu-latest
+      container: mbedos/mbed-os-env:latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+
+        - name: Install dependencies
+          run: |
+            apt-get update && apt-get install -y valgrind gcovr
+            pip3 install virtualenv
+
+        - name: Run unit tests
+          run: ./tests/unittests.sh
+
+        - name: Upload coverage result
+          uses: actions/upload-artifact@v2
+          with:
+            name: gcovr coverage result
+            path: tests/coverage.html
+
+  run-integration-tests:
+    name: Run integration tests
+    runs-on: ubuntu-latest
+    container: mbedos/mbed-os-env:latest
```

Comment:
if the cmake here is too old maybe try mbedos/mbed-os-env-cmake?

Reply:
mbed-tools issues resolved in 7f221b0c8059b5b273aae43520c89a9a9ca620a1 and 88394b1f78f3a1e7be6595eac0aa7dd0e37cf20a